crates/bevy_gizmos/src/lib.rs

```diff
@@ -134,108 +142,67 @@ impl Plugin for GizmoPlugin {
     }
 }
 
-/// A [`Resource`] that stores configuration for gizmos.
-#[derive(Resource, Clone)]
-pub struct GizmoConfig {
-    /// Set to `false` to stop drawing gizmos.
-    ///
-    /// Defaults to `true`.
-    pub enabled: bool,
-    /// Line width specified in pixels.
-    ///
-    /// If `line_perspective` is `true` then this is the size in pixels at the camera's near plane.
+/// A trait adding `init_gizmo_config<T>()` to the app
+pub trait AppGizmoBuilder {
```

Comment:
I'm quite new to rust and big projects and would love to understand why this should be moved to its own module. For me, it seems to be related to the plugin build code above. Thanks!

Reply:
The general pattern in rust is to have a tinny `lib.rs`. In bevy, it usually only consists of re-exports (such as the prelude) and the root `impl Plugin for CratePlugin` definition.

The whole `bevy_gizmo` crate started small enough that it could fit in a single file. But as we add features, it is good to split up. I think it's time to split it, just generally.

In the case of `AppGizmoBuilder`, some crate authors define extension traits in a separate module, so it would be fit to move to its own module.

Reply:
I moved the aabb code into a separate module as I changed how additional gizmo configs should be added. But I think splitting the whole `lib.rs` should be done in another PR.